app/org/sagebionetworks/bridge/services/ActivityEventService.java

```diff
@@ -53,7 +54,11 @@ public void publishCustomEvent(Study study, String healthCode, String eventKey,
                 .withObjectType(ActivityEventObjectType.CUSTOM)
                 .withObjectId(eventKey)
                 .withTimestamp(timestamp).build();
-        activityEventDao.publishEvent(event);
+        
+        if (activityEventDao.publishEvent(event)) {
+            // Create automatic events, as defined in the study
+            createAutomaticCustomEvents(study, healthCode, event);
+        }
```

Comment:
We have a circular dependency here. createAutomaticCustomEvents() calls publishCustomEvents() which calls createAutomaticCustomEvents(). It would be difficult to construct a custom event config that causes an infinite recursion here.

Reply:
Now calling the DAO directly rather than back into the service for the creation of these automatic custom events.